contracts/actions/mcd/McdSubProxy.sol

```diff
@@ -0,0 +1,155 @@
+            _validateSubData(_subData);
+
+            StrategySub memory boostSub = formatBoostSub(_subData);
+
+            // if we don't have a boost bundleId, create one
+            if (_subId2 == 0) {
+                SubStorage(SUB_STORAGE_ADDR).subscribeToStrategy(boostSub);
+            } else {
+                SubStorage(SUB_STORAGE_ADDR).updateSubData(_subId2, boostSub);
+                SubStorage(SUB_STORAGE_ADDR).activateSub(_subId2);
+            }
+        } else {
+            if (_subId2 != 0) {
+                SubStorage(SUB_STORAGE_ADDR).deactivateSub(_subId2);
+            }
+        }
+    }
+
+    /// @notice Activates Repay sub and if exists a Boost sub
+    function activateSub(
+        uint32 _subId1,
+        uint32 _subId2
+    ) public {
+        SubStorage(SUB_STORAGE_ADDR).activateSub(_subId1);
+
+        if (_subId2 != 0) {
+            SubStorage(SUB_STORAGE_ADDR).activateSub(_subId2);
+        }
+    }
+
+    /// @notice Deactivates Repay sub and if exists a Boost sub
+    function deactivateSub(
+        uint32 _subId1,
+        uint32 _subId2
+    ) public {
+        SubStorage(SUB_STORAGE_ADDR).deactivateSub(_subId1);
+
+        if (_subId2 != 0) {
+            SubStorage(SUB_STORAGE_ADDR).deactivateSub(_subId2);
+        }
+    }
+
+
+    ///////////////////////////////// HELPER FUNCTIONS /////////////////////////////////
+
+    function _validateSubData(McdSubData memory _subData) internal pure {
+        if (_subData.minRatio > _subData.maxRatio) {
+            revert WrongSubParams(_subData.minRatio, _subData.maxRatio);
+        }
+
+        if ((_subData.maxRatio - RATIO_OFFSET) < _subData.targetRatioRepay) {
+            revert RangeTooClose(_subData.maxRatio, _subData.targetRatioRepay);
+        }
+
+        if ((_subData.minRatio + RATIO_OFFSET) > _subData.targetRatioBoost) {
+            revert RangeTooClose(_subData.minRatio, _subData.targetRatioBoost);
+        }
+    }
+
+    /// @notice Formats a StrategySub struct to a Repay bundle from the input data of the specialized aave sub
+    function formatRepaySub(McdSubData memory _user) public pure returns (StrategySub memory repaySub) {
+        repaySub.strategyOrBundleId = REPAY_BUNDLE_ID;
+        repaySub.isBundle = true;
+
+        // format data for ratio trigger if currRatio < minRatio = true
+        bytes memory triggerData = abi.encode(_user.vaultId, uint256(_user.minRatio), uint8(RatioState.UNDER));
+        repaySub.triggerData =  new bytes[](1);
+        repaySub.triggerData[0] = triggerData;
+
+        repaySub.subData =  new bytes32[](2);
+        repaySub.subData[0] = bytes32(_user.vaultId);
+        repaySub.subData[1] = bytes32(uint256(_user.targetRatioRepay));
+        
+    }
+
+    /// @notice Formats a StrategySub struct to a Boost bundle from the input data of the specialized aave sub
+    function formatBoostSub(McdSubData memory _user) public pure returns (StrategySub memory repaySub) {
```

Comment:
StrategySub memory repaySub -> boostSub and comments are saying aave instead of mcd